packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -267,24 +335,131 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Unregister all jobs related to the given snapId.
+   * Schedule a background event.
+   *
+   * @param backgroundEventWithoutId - Background event.
+   * @returns An id representing the background event.
+   */
+  scheduleBackgroundEvent(
+    backgroundEventWithoutId: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+  ) {
+    const event = {
+      ...backgroundEventWithoutId,
+      id: nanoid(),
+      scheduledAt: new Date().toISOString(),
+    };
+    this.#setUpBackgroundEvent(event);
+    this.update((state) => {
+      state.events[event.id] = castDraft(event);
+    });
+
+    return event.id;
+  }
+
+  /**
+   * Cancel a background event.
+   *
+   * @param id - The id of the background event to cancel.
+   * @param origin - The origin making the cancel call.
+   * @throws If the event does not exist.
+   */
+  cancelBackgroundEvent(id: string, origin: string) {
+    assert(
+      this.state.events[id],
+      `A background event with the id of "${id}" does not exist.`,
+    );
+
+    assert(
+      this.state.events[id].snapId === origin,
+      'Only the origin that scheduled this event can cancel it',
+    );
+
+    const timer = this.#timers.get(id);
+    timer?.cancel();
+    this.#timers.delete(id);
+    this.#snapIds.delete(id);
+    this.update((state) => {
+      delete state.events[id];
+    });
+  }
+
+  /**
+   * A helper function to handle setup of the background event.
+   *
+   * @param event - A background event.
+   */
+  #setUpBackgroundEvent(event: BackgroundEvent) {
+    const date = new Date(event.date);
```

Comment:
We should decide on what timezone to use when timezone is omitted: https://github.com/MetaMask/SIPs/pull/152

This seems to default to local.

Reply:
Oh right hmm, I'm for requiring timezone info. There's some ambiguity with local timezone so I think it's better to just require timezone info.